srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.md

```diff
@@ -599,171 +599,201 @@ frames. If only sending audio, the RECOMMENDED frequency is every
+
+[] Messages protected using DTLS application traffic keys
+
+<> Messages protected using the EKTKey and EKT cipher
+
+|| Messages protected using the SRTP Master Key sent in
+   a Full EKT Tag
+~~~
+
+When used in multi-party SRTP sessions, the DTLS Server may be a different
+entity than the source of the SRTP packets.  If the DTLS-SRTP server performs a
+handshake with each DTLS-SRTP endpoint to set a common EKT key, then each SRTP
+sender can use EKT to inform other participants of the keys it is using to
+send.  This avoids the need for many individual DTLS handshakes among the
+endpoints, at the cost of preventing endpoints from directly authenticating one
+another.
+
+~~~
+Client A                 Server                 Client B
+
+    <----DTLS Handshake---->
+    <--------EKTKey---------
+                            <----DTLS Handshake---->
+                            ---------EKTKey-------->
+
+    -------------SRTP Packet + EKT Tag------------->
+    <------------SRTP Packet + EKT Tag--------------
+~~~
+
+
+### Negotiating an EKT Cipher
+
+To indicate its support for EKT, a DTLS-SRTP client includes in its ClientHello
+an extension of type `supported_ekt_ciphers` listing the EKT ciphers the client
+supports.  If the server agrees to use EKT, then it includes a
+`supported_ekt_ciphers` extension in its ServerHello containing a cipher
+selected from among those advertised by the client.
+
+The `extension_data` field of this extension contains an "EKTCipher" value,
+encoded using the syntax defined in [@!RFC5246]:
+
+~~~
 enum {
-  ekt_key(0),
-  ekt_key_ack(1),
-  ekt_key_error(254),
-  (255)
-} EKTMessageType;
+  reserved(0),
+  aeskw_128(1),
+  aeskw_256(2),
+} EKTCipherType;
 
 struct {
-  EKTMessageType ekt_message_type;
-  select (EKTMessage.ekt_message_type) {
-  case ekt_key:
-    EKTKey;
-  } message;
-} EKTMessage;
+    select (Handshake.msg_type) {
+        case client_hello:
+            EKTCipherType client_shares<1..255>;
+
+        case server_hello:
+            EKTCipherType selected_cipher;
+    };
+} EKTCipher;
 ~~~
-Figure: Additional TLS Data Structures
-
-If a DTLS client includes srtp\_ekt\_key\_transport in its ClientHello,
-then a DTLS server that supports this extensions will includes
-srtp\_ekt\_key\_transport in its ServerHello message. If a DTLS client
-includes srtp\_ekt\_key\_transport in its ClientHello, but does not
-receive srtp\_ekt\_key\_transport in the ServerHello, the DTLS client
-MUST NOT send DTLS EKTMessage messages. Also, the
-srtp\_ekt\_key\_transport in the ServerHello MUST select one and only
-one EKTCipherType from the list provided by the client in the
-srtp\_ekt\_key\_transport in the ClientHello.
-
-When a DTLS client sends the srtp\_ekt\_key\_transport in its
-ClientHello message, it MUST include the SupportedEKTCiphers as the
-extension\_data for the extension, listing the EKTCipherTypes the
-client is willing to use in preference order, with the most preferred
```

Comment:
This is the preferred order text that we lost. I suspect we need something like it in the new text ( see other comment )